tests/SpecTests/DocumentsMatchConstraint.php

```diff
@@ -421,6 +401,11 @@ private function doToString()
         return 'matches ' . $this->exporter()->export($this->value);
     }
 
+    private static function isNumeric($value): bool
+    {
+        return is_int($value) || is_float($value) || $value instanceof Int64;
```

Comment:
`prepareBSON` has logic to "Convert Int64 objects to integers on 64-bit platforms for compatibility reasons." Was that sufficient to allow flexible comparisons of Int64 objects with integers and floats? Mainly asking because I don't believe we have a test for this.

Int64 is conditionally used in DocumentsMatchConstraintTest only when the platform is 32-bit.

This probably makes no difference, but the fact that you reference the class here made me ask.

Reply:
I shamelessly copied the logic from the unified spec tests where this was used for flexible numeric comparisons.